RELEASE/scripts/autoscend/auto_bedtime.ash

```diff
@@ -719,6 +719,21 @@ boolean doBedtime()
 	{
 		auto_log_info("Please consider genie wishing for an orcish frat boy spy (You want Frat Warrior Fatigues).", "blue");
 	}
+	
+	if(item_amount($item[Infinite BACON Machine]) > 0 && !get_property("_internetViralVideoBought").to_boolean() && !can_interact())
+	{
+		boolean hasFireworks = (item_amount($item[Clan VIP Lounge Key]) > 0 && get_property("_fireworksShop").to_boolean());
```

Comment:
mr2021 has a function for this, but take a look at my next comment before changing this as it may be moot.
```suggestion
		boolean hasFireworks = have_fireworks_shop();
```